client/lib/gsuite/new-users.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import emailValidator from 'email-validator';
+import i18n from 'i18n-calypso';
+import { includes, mapValues } from 'lodash';
+
+// exporting these in the big export below causes trouble
+export interface GSuiteNewUserField {
+	value: string;
+	error: string | null;
+}
+
+export interface GSuiteNewUser {
+	domain: GSuiteNewUserField;
+	mailBox: GSuiteNewUserField;
+	firstName: GSuiteNewUserField;
+	lastName: GSuiteNewUserField;
+}
+
+export interface GSuiteProductUser {
+	firstname: string;
+	lastname: string;
+	email: string;
+}
+
+const removePreviousErrors = ( { value }: GSuiteNewUserField ): GSuiteNewUserField => ( {
+	value,
+	error: null,
+} );
+
+const requiredField = ( { value, error }: GSuiteNewUserField ): GSuiteNewUserField => ( {
+	value,
+	error:
+		! error && ( ! value || '' === value ) ? i18n.translate( 'This field is required.' ) : error,
+} );
+
+const sixtyCharacterField = ( { value, error }: GSuiteNewUserField ): GSuiteNewUserField => ( {
+	value,
+	error:
+		! error && 60 < value.length
+			? i18n.translate( "This field can't be longer than 60 characters." )
```

Comment:
We could extract the number of characters to be able to reuse this string in a different context where another limit would apply.

Reply:
fixed